packages/core/src/contrib/Queries.ts

```diff
@@ -368,6 +415,42 @@ export const shapeDistance = (s1: Shape<ad.Num>, s2: Shape<ad.Num>): ad.Num => {
       toPt(s2.start.contents),
       toPt(s2.end.contents)
     );
+  } else if (t1 === "Polyline" && t2 === "Circle") {
+    return shapeDistanceCirclePolyline(
+      s2.center.contents,
+      s2.r.contents,
+      s1.points.contents
+    );
+  } else if (t2 === "Polyline" && t1 === "Circle") {
+    return shapeDistanceCirclePolyline(
+      s1.center.contents,
+      s1.r.contents,
+      s2.points.contents
+    );
+  } else if (t1 === "Polyline" && isRectlike(s2)) {
+    const bbox = bboxFromShape(s2);
+    const corners = BBox.corners(bbox);
```

Comment:
There is a function called `bboxPts(s)` that computes the bounding box of `s` and returns the four points of the bounding box (top-right, top-left, bottom-left, bottom-right). By the way, this function apparently assumes that the rectangles have zero rotation value - maybe that's something that can be written in the documentation?

Reply:
@liangyiliang The function `bboxPts()` just calls `BBox.corners`, as is done here, right?

```
export const bboxPts = (bbox: BBox.BBox): ad.Pt2[] => {
  const { topLeft, topRight, bottomLeft, bottomRight } = BBox.corners(bbox);
  return [topRight, topLeft, bottomLeft, bottomRight];
};
```

I don't see a version that takes a shape directly as an argument.  Most places it is used like `      bboxPts(bboxFromShape(s))`.  Wouldn't it therefore be better to everywhere write

```
BBox.corners(bboxFromShape( s ))
```

rather than

```
bboxPts(bboxFromShape(s))
```
?

Basically I don't understand the point of the `bboxPts` function (esp. since there are no comments).  Is it just to do a type conversion?

Reply:
By the way, a lot of these things (like how corners are accessed, whether rect-line functions support rotations, etc.) are not introduced in my PR; I'm just extending to Polylines using existing functions.  I would prefer if we address things like rotations in a future PR, since this one is already quite complicated.